modules/offlinemaps/src/main/java/com/sygic/maps/offlinemaps/regions/RegionsFragment.kt

```diff
@@ -0,0 +1,111 @@
+import com.sygic.maps.offlinemaps.base.NavigationFragment
+import com.sygic.maps.offlinemaps.databinding.FragmentRegionsBinding
+import com.sygic.sdk.map.MapLoader
+import kotlinx.android.synthetic.main.fragment_regions.*
+
+class RegionsFragment : NavigationFragment<RegionsViewModel>() {
+    override val viewModel by viewModels<RegionsViewModel>()
+
+    private val regionsAdapter: RegionListAdapter
+        get() = regionList.adapter as RegionListAdapter
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val binding = FragmentRegionsBinding.inflate(inflater)
+        binding.lifecycleOwner = this
+        binding.adapter = RegionListAdapter()
+        binding.viewModel = viewModel
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initArguments(requireArguments())
+
+        val countryViewHolder = CountryEntryViewHolder(countryLayout, false)
+        countryViewHolder.setPrimaryActionClickListener {
+            viewModel.onCountryPrimaryActionClicked()
+        }
+
+        viewModel.countryObservable.observe(viewLifecycleOwner, Observer { countries ->
+            val countryHolder = countries[viewModel.country]!!
+            viewModel.countryHolder = countryHolder
+            countryViewHolder.bind(countryHolder)
+            progressIndicator.visibility = View.GONE
+        })
+
+        viewModel.regionObservable.observe(viewLifecycleOwner, Observer { regions ->
+            viewModel.countryHolder?.let { holder ->
+                val regionList = holder.country.details.regions.map { regions[it]!! }.filter {
+                    if (viewModel.installed) {
+                        it.status == MapLoader.MapStatus.Loaded || it.status == MapLoader.MapStatus.Installed
+                    } else {
+                        true
+                    }
+                }
+                regionsAdapter.submitList(regionList)
+            }
+        })
+
+        viewModel.notifyMapChangedObservable.observe(viewLifecycleOwner, Observer {
+            if (it.first == viewModel.country) {
+                countryViewHolder.updateFromStatus(it.second, 0)
+            } else {
+                regionsAdapter.updateStatus(it.first, it.second)
+            }
+        })
+
+        viewModel.mapInstallProgressObservable.observe(viewLifecycleOwner, Observer {
+            if (it.first == viewModel.country) {
+                countryViewHolder.updateProgress(it.second)
+            } else {
+                regionsAdapter.updateProgress(it.first, it.second)
+            }
+        })
+
+        viewModel.loadMaps()
+    }
+
+    private fun initArguments(arguments: Bundle) {
+        viewModel.country = arguments.getString(COUNTRY_KEY) ?: throw IllegalArgumentException("Must pass country ISO in arguments")
```

Comment:
minor vec, ale toto sa da pekne spravit ako
      ``  viewModel.country = requireNotNull(arguments.getString(COUNTRY_KEY)) { "Must pass country ISO in arguments" }``

Ale dobre je to aj tak, ako je to teraz :-)

Reply:
Upravil som aby to vyzeral krajsie :)